### DISCUSSION TITLE
Vulnerability fix PRs follow GitHub alerts and will miss a package file if GitHub misses it

### DISCUSSION THREAD

**Discussion by klippx** [renovatebot/renovate#21577] [Request Help] [answered]
### What version of Renovate do you run?

29.36.2

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am using `config:js-app` which contains `:pinAllExceptPeerDependencies`

As a result, renovate creates a PR called `Pin dependency type-graphql to 0.0.2 [SECURITY]` which touches the following files:

```sh
 packages/**/package.json
 yarn.lock
```

However, these are also present in root `package.json` in the `resolutions` section, and they need to match. So basically it feels like the root package.json is ignored, but it needs to be pinned there as well!

Wanted end result: I need these files updated / pinned version:

```sh
 packages/**/package.json
 package.json
 yarn.lock
```

### What you tried so far.

This is my config:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "username": "renovate-release",
  "gitAuthor": "Renovate Bot <bot@renovateapp.com>",
  "onboarding": false,
  "platform": "github",
  "repositories": ["my-organisation/graphql-federation-mono"],
  "labels": ["dependencies"],
  "extends": [
    "config:js-app",
    ":disableDependencyDashboard",
    ":disableMajorUpdates"
  ]
}
```

I have also tried adding
```json
  "packageRules": [
    {
      "matchDepTypes": ["resolutions"],
      "rangeStrategy": "pin"
    }
  ]
```

But to no avail. It doesn't touch the root package.json.

**Comment by rarkins** [comment]
If this is a vulnerability remediation then it's driven by GitHub's vulnerability alerts. If the vulnerability alerts don't detect the root package resolution, then Renovate won't remediate it

**Reply by klippx** [reply]
Allright, but it happens for all PRs though, not the same issue but a related one

Here are two examples:

### `Update dependency @types/validator to v13.7.15`

In this PR, dependabot updated `resolutions` field in root package.json only, from `13.7.1` to `13.7.15` like advertised. But in `packages/**/package.json`, where this packages was pinned in `devDependencies` as `13.7.1`, it was not updated. So I had to make a separate commit to help the branch out. 

### Update dependency @types/graphql-fields to v1.3.5

In this PR, dependabot updated `resolutions` field in root package.json only, from `1.3.4` to `1.3.5` like advertised. But in `packages/**/package.json`, where this packages was pinned in `devDependencies` as `1.3.4`, it was not updated. So I had to make a separate commit to help the branch out. 

This is about to get a bit tedious since I reckon we could have about 50 more PRs to go through.

**Comment by HonkingGoose** [comment]
> ### What version of Renovate do you run?
> 
> 29.36.2

That version is from December 2021. If you're _really_ using that old version: please update to the latest version of Renovate. 😉

It may not fix your problem, but you'll still get new features and updated Renovate-internal dependencies.

**Reply by klippx** [reply, marked as answer]
I will. But as I just adopted `"config:js-app"` which came with a lot of config changes I didn't want to stir the pot too much.

In the end my solution was to get rid of `resolutions` in `package.json`, we didn't need it and it plays better with renovate.

**Reply by HonkingGoose** [reply]
By the way, we now publish expanded release notes with Developers Commentary for the latest major releases. We have these notes for `33`, `34` and `35`. I recommend you read those notes when you're going to update Renovate.

https://docs.renovatebot.com/release-notes-for-major-versions/

**Reply by klippx** [reply]
When is `36` coming? Am I on alpha if I took `36`?

**Reply by HonkingGoose** [reply]
I'm confused, we should **not** be publishing any `36` version yet. Where did you find that `36` version, do you have a link?

We have a `v36` branch to prepare things: https://github.com/renovatebot/renovate/tree/v36 but we don't want anybody to use that version. We'll release `v36` when it's ready for the public.

---

Edit: I was totally wrong here, the GitHub Action version no longer matches the `renovatebot/renovate` version. See my comment lower in this thread.

**Reply by klippx** [reply]
<img width="1124" alt="Screenshot 2023-04-20 at 14 36 00" src="https://user-images.githubusercontent.com/2785359/233367732-bd54e361-1baa-40a3-9bc4-0b40278ba2e2.png">

https://github.com/marketplace/actions/renovate-bot-github-action

It seems like it is the official and recommended / current version in the marketplace? Maybe it is me that is confused? 😓

**Reply by HonkingGoose** [reply]
> It seems like it is the official and recommended / current version in the marketplace?

Correct. The GitHub Action Renovate is coming from our `renovatebot` organization, so it's official. And I see _why_ we already have a `v36` of the GitHub Action: [^1]

> This action no longer follows renovate versions, use [`renovate-version`](https://github.com/renovatebot/github-action#renovate-version) to pin the renovate version.

And this issue explains why we decoupled:

- https://github.com/renovatebot/github-action/issues/689

> Maybe it is me that is confused? 😓

Actually I was confused! 😄 I did not know about this change. I assumed that all our official spin-offs from the main Renovate program would follow the main versioning. You can select what version of Renovate you want the Action to run by setting the `renovate-version`. [^2]

[^1]: https://github.com/renovatebot/github-action/releases/tag/v36.0.0
[^2]: https://github.com/HonkingGoose/github-action#renovate-version

**Reply by klippx** [reply]
With this in mind, do you recommend that users simply pick `uses: renovatebot/github-action@v36.0.0` and not pin `renovate-version`?

I am guessing the idea is to get the latest non-breaking 36.x version. And if you make a new major, we simply upgrade `uses: renovatebot/github-action@v37.0.0`. Is that it?

> use [renovate-version](https://github.com/renovatebot/github-action#renovate-version) to pin the renovate version.

The only reason I am in doubt is this ☝️  I would only use that if you accidentally released something that didn't work for our project, does that sound about right? 🤔
